### PR TITLE
Add database set/query and cap query

### PR DESCRIPTION
**Pull request by yuvalavn**

### CODE REVIEW

File: drivers/regex/mlx5/mlx5_regex.c [hunk 1]
```diff
@@ -3,3 +3,15 @@
  */
 
 #include "mlx5_regex.h"
+
+static int mlx5_regex_is_supported(struct ibv_context *ibv_ctx)
```
Comment (paravmellanox) [inline drivers/regex/mlx5/mlx5_regex.c]: do we have the bool data type in  dpdk?
if so, please return as bool

**Review by paravmellanox — COMMENTED** [review]
please update the commit log of what each stop, resume apis do and when to use it, limitation etc.

File: drivers/regex/mlx5/Makefile [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+LDLIBS += -lrte_eal -lrte_mbuf -lrte_mempool -lrte_ring
+LDLIBS += -lrte_kvargs
+LDLIBS += -lrte_bus_pci
+
+# A few warnings cannot be avoided in external headers.
+CFLAGS += -Wno-error=cast-qual
+
+EXPORT_MAP := rte_pmd_mlx5_version.map
+
+# memseg walk is not part of stable API
```
Comment (paravmellanox) [inline drivers/regex/mlx5/Makefile:34]: can we avoid it?

File: drivers/regex/mlx5/mlx5_regex.c (hunk 1 above)
Comment (orikam) [reply drivers/regex/mlx5/mlx5_regex.c]: There are some cases of bool usage in DPDK but not many
I prefer int so we can use it as error code,

**Review by paravmellanox — COMMENTED** [review]
@orikam 
it is a static function so not so useful on error code.
and error code on false can be easily converted to ENOSUPP

File: drivers/regex/mlx5/Makefile (hunk 2 above)
Comment (yuvalavn) [reply drivers/regex/mlx5/Makefile:34]: I relied on the mlx5 PMD here. assuming we need the same libs.
So maybe forgot to remove some things.

Comment (yuvalavn) [reply drivers/regex/mlx5/Makefile:34]: Yes, it is needed

**Review by orikam — APPROVED** [review]

**Pull request merged**